kie-server-parent/kie-server-services/kie-server-services-openshift/src/test/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepositoryRegularTest.java

```diff
@@ -394,5 +405,32 @@ public void testKieContainerUpdateDuringRolloutAllowed() {
         assertTrue(repo.isKieContainerUpdateDuringRolloutAllowed(cfmContainer, stateWithStopped));
         // Does not allow state transition from STOPPED to STARTED as it is not necesscary
         assertFalse(repo.isKieContainerUpdateDuringRolloutAllowed(cfmStopped, state));
-}
+    }
+
+    @Test
+    public void testLabelSelector() {
+        LabelSelector selector = repo.getKieServerCMLabelSelector(client);
+        assertNotNull(selector);
+        // By default, global discovery is disabled, meaning that it must have an 'application' label in the selector
+        boolean foundAppNameLabel = false;
+        for (LabelSelectorRequirement selReq : selector.getMatchExpressions()) {
+            if (selReq.getKey() == KieServerStateCloudRepository.CFG_MAP_LABEL_APP_NAME) {
+                foundAppNameLabel = true;
+                assertEquals("In", selReq.getOperator());
+                assertTrue(selReq.getValues().contains(TEST_APP_NAME));
+            }
+            if (selReq.getKey() == KieServerStateCloudRepository.CFG_MAP_LABEL_NAME) {
+                assertEquals("In", selReq.getOperator());
+                assertTrue(selReq.getValues().contains(KieServerStateCloudRepository.CFG_MAP_LABEL_VALUE_USED));
+                assertTrue(selReq.getValues().contains(KieServerStateCloudRepository.CFG_MAP_LABEL_VALUE_IMMUTABLE));
+            }
+        }
+        assertTrue(foundAppNameLabel);
+                
+        System.setProperty(KieServerStateCloudRepository.CFG_KIE_CONTROLLER_OCP_GLOBAL_DISCOVERY_ENABLED, "true");
+        assertTrue(repo.getKieServerCMLabelSelector(client).getMatchExpressions().stream().noneMatch(
+            selReq -> selReq.getKey() == KieServerStateCloudRepository.CFG_MAP_LABEL_APP_NAME));
+        System.clearProperty(KieServerStateCloudRepository.CFG_KIE_CONTROLLER_OCP_GLOBAL_DISCOVERY_ENABLED);
```

Comment:
Please put this into final statement of the try clause containing previous line or move it to `@After` method, so the property is cleared also in case of assertion failure

Reply:
Updated as per suggestion